src/literalset.rs

```diff
@@ -40,6 +40,18 @@ impl<'l> TrimmedLiteralRef<'l> {
     pub fn as_ref(&self) -> &TrimmedLiteral {
         self.reference
     }
```

Comment:
I think truncation is not the right term, it indicates a change of state, it's more of an `extraction` or `sub`? We could also just take a `Range` as argument and use that and check for the range bounds to be within the the string length.
Remember the `string[char_index]` already works on chars iirc :)

Reply:
I agree with you. I have replaced with sub. I rewrote it using the ranges + edge cases. Let me know what you think :)